dice.cpp

```diff
@@ -0,0 +1,33 @@
+#include "dice.h"
+
+Dice::Dice(){
+    sum =0;
+}
+
+Dice::~Dice(){}
+
+void Dice::displayResults(){
+    std::cout << "Results of "<< results.size() << "d" << diceSides << " rolls:" << std::endl;
+    for(auto i : results){
+        std::cout << i << ":";
+        sum += i;   //Sum showing up as 140737488349248 in debugger
+    }
+    std::cout << std::endl << "The total of all rolls: " << sum << std::endl;
+
+    //After displaying results we can discard
+    sum = 0;
+    results.clear();
+}
+
+void Dice::roll(int numberDice = 1, int diceSides = 1){
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_int_distribution<> dis(1, diceSides);
```

Comment:
You should probably do a check to make sure diceSides has an appropriate value before using it.